tests/test_run_releasenotes_check.py

```diff
@@ -0,0 +1,52 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+import unittest
+from typing import Any
+from unittest.mock import patch
+
+import pytest
+
+from run_releasenotes_check import main
+
+
+class TestRunReleaseNotesCheck(unittest.TestCase):
+    @pytest.fixture(autouse=True)
+    def _capfd(self, capfd: Any) -> None:
+        self.capfd = capfd
+
+    @patch("argparse._sys.argv", ["run_releasenotes_check.py", "--help"])
+    def test_usage(self) -> None:
+        with self.assertRaises(SystemExit):
+            main()
+
+        out, _ = self.capfd.readouterr()
+        self.assertTrue(out.startswith("usage:"))
+
+    OPENSEARCH_MANIFEST = os.path.realpath(
+        os.path.join(
+            os.path.dirname(__file__),
+            "..",
+            "manifests",
+            "templates",
+            "opensearch",
+            "2.x",
+            "manifest.yml"
+        )
+    )
+
+    @patch('subprocess.check_call')
+    @patch("argparse._sys.argv", ["run_releasenotes_check.py", "check", OPENSEARCH_MANIFEST, "--date", "2022-07-26"])
+    @patch("run_releasenotes_check.ReleaseNotes")
+    def test_main(self, *mocks: Any) -> None:
+        main()
+
+    @patch('subprocess.check_call')
+    @patch("argparse._sys.argv", ["run_releasenotes_check.py", "check", OPENSEARCH_MANIFEST, "--date", "2022-07-26", "--output", "test.md"])
+    @patch("run_releasenotes_check.ReleaseNotes")
+    def test_main_with_save(self, *mocks: Any) -> None:
+        main()
```

Comment:
This only tests that `main()` doesn't throw any errors. You need to assert that it tried to write the output file, aka that `writer.dump` was called.

Reply:
Added more assertions.